daemon/mgr/container.go

```diff
@@ -364,6 +367,32 @@ func (cm *ContainerManager) List(ctx context.Context) ([]*types.ContainerInfo, e
 	return cis, nil
 }
 
+// Rename rename a exist container
+func (cm *ContainerManager) Rename(ctx context.Context, name string, newName string) error {
+	var (
+		ci  *types.ContainerInfo
+		err error
+	)
+
+	if cm.NameToID.Get(newName).Exist() {
+		return errors.New("The newName already exists")
```

Comment:
Try to use pkg https://github.com/alibaba/pouch/blob/master/pkg/httputils/http_error.go to wrap the error with status code 409 which is defined in your swagger.yml.